--- yp.c	2004-05-31 23:01:18.000000000 +0200
+++ yp.c.oden	2012-05-06 09:29:38.479033731 +0200
@@ -41,7 +41,7 @@ PHP_YP_API php_yp_globals yp_globals;
 
 /* }}} */
 
-function_entry yp_functions[] = {
+zend_function_entry yp_functions[] = {
 	PHP_FE(yp_get_default_domain, NULL)
 	PHP_FE(yp_order, NULL)
 	PHP_FE(yp_master, NULL)
@@ -94,7 +94,7 @@ PHP_FUNCTION(yp_get_default_domain)
    Returns the order number or false */
 PHP_FUNCTION(yp_order)
 {
-	pval **domain, **map;
+	zval **domain, **map;
 
 #if SOLARIS_YP
 	unsigned long outval;
@@ -122,7 +122,7 @@ PHP_FUNCTION(yp_order)
    Returns the machine name of the master */
 PHP_FUNCTION(yp_master)
 {
-	pval **domain, **map;
+	zval **domain, **map;
 	char *outname;
 
 	if((ZEND_NUM_ARGS() != 2) || zend_get_parameters_ex(2,&domain,&map) == FAILURE) {
@@ -145,7 +145,7 @@ PHP_FUNCTION(yp_master)
    Returns the matched line or false */
 PHP_FUNCTION(yp_match)
 {
-	pval **domain, **map, **key;
+	zval **domain, **map, **key;
 	char *outval;
 	int outvallen;
 
@@ -170,7 +170,7 @@ PHP_FUNCTION(yp_match)
    Returns the first key as array with $var[$key] and the the line as the value */
 PHP_FUNCTION(yp_first)
 {
-	pval **domain, **map;
+	zval **domain, **map;
 	char *outval, *outkey;
 	int outvallen, outkeylen;
 
@@ -198,7 +198,7 @@ PHP_FUNCTION(yp_first)
    Returns an array with $var[$key] and the the line as the value */
 PHP_FUNCTION(yp_next)
 {
-	pval **domain, **map, **key;
+	zval **domain, **map, **key;
 	char *outval, *outkey;
 	int outvallen, outkeylen;
 
@@ -270,7 +270,7 @@ static int php_foreach_all (int instatus
    Traverse the map and call a function on each entry */
 PHP_FUNCTION(yp_all)
 {
-	pval **domain, **map, **php_callback;
+	zval **domain, **map, **php_callback;
 	struct ypall_callback callback;
 
 	if((ZEND_NUM_ARGS() != 3) || zend_get_parameters_ex(3,&domain,&map,&php_callback) == FAILURE) {
@@ -325,7 +325,7 @@ static int php_foreach_cat (int instatus
    Return an array containing the entire map */
 PHP_FUNCTION(yp_cat)
 {
-	pval **domain, **map;
+	zval **domain, **map;
 	struct ypall_callback callback;
 
 	if((ZEND_NUM_ARGS() != 2) || zend_get_parameters_ex(2,&domain,&map) == FAILURE) {
@@ -360,7 +360,7 @@ PHP_FUNCTION(yp_errno)
    Returns the corresponding error string for the given error code */
 PHP_FUNCTION(yp_err_string)
 {
-	pval **error;
+	zval **error;
 	char *string;
 
 	if((ZEND_NUM_ARGS() != 1) || zend_get_parameters_ex(1,&error) == FAILURE) {
